pyspecdata/lmfitdata.py

```diff
@@ -303,18 +280,31 @@ def fit(self):
         # }}}
         return self
 
-    def run_lambda(self, pars):
-        """actually run the lambda function we separate this in case we want
-        our function to involve something else, as well (e.g. taking a Fourier
-        transform)"""
-        logging.debug(strm(self.getaxis(j) for j in self.variable_names))
-        return self.fitfunc_multiarg_v2(
-            *(self.getaxis(j) for j in self.variable_names), **pars.valuesdict()
-        )
+    def run_lambda(self, pars, thistaxis = None):
+        """actually run the lambda function that calculates the model data.
+        Note that the name of the variable along which the model data is calculated
+        (as opposed to "parameter" is set by variable_names parameter).
+
+        .. note::
+            we had separates this in case we want
+            our function to involve something else, as well (e.g. taking a Fourier
+            transform).  Unknown if there are still two steps in this way.
+
+        """
+        if len(self.variable_names) > 1:
+            print("We are only allowing for one variable for now")
+        elif thistaxis is None:
+            return self.fitfunc_multiarg_v2(
+                    *(self.getaxis(j) for j in self.variable_names), **pars.valuesdict())
+        else:    
+            logging.debug(strm(self.getaxis(j) for j in self.variable_names))
+            return self.fitfunc_multiarg_v2(
+                    thistaxis, **pars.valuesdict()
+            )
```

Comment:
yes, this is good!

Reply:
let's modify, though, since we are thinking about this.
`thistaxis is None` to `len(kwargs) == 0` then in else, 
first argument is not `thistaxis` but `*(kwargs[j] if j in kwargs.keys() else self.getaxis(j) for j in self.variable_names)`

happy to talk about that to make sure it makes sense.

Reply:
and `thistaxis=None` gets replaced by `**kwargs`